src/dbnode/encoding/m3tsz/timestamp_encoder.go

```diff
@@ -182,6 +185,12 @@ func (enc *TimestampEncoder) writeAnnotation(stream encoding.OStream, ant ts.Ann
 	stream.WriteBytes(buf[:annotationLength])
 	stream.WriteBytes(ant)
 
+	if enc.PrevAnnotationChecksum != emptyAnnotationChecksum {
+		// NB: current assumption is that in the most cases no annotation writes should happen and that
```

Comment:
`no annotation writes should happen` I think is a bit misleading - we expect to have one annotation per block on every time series. I think the comment should speak of rewrites specifically, and not writes in general.

Reply:
https://github.com/m3db/m3/pull/3565/commits/3833ce9083a885a585ab43f0652e06bdbc03a8d7